fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/DateTimeSortRequestHandler.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.handlers;
+
+import com.yahoo.bard.webservice.data.metric.mappers.DateTimeSortMapper;
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery;
+import com.yahoo.bard.webservice.web.DataApiRequest;
+import com.yahoo.bard.webservice.web.responseprocessors.MappingResponseProcessor;
+import com.yahoo.bard.webservice.web.responseprocessors.ResponseProcessor;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Injects the result set mapper for dateTime based sort if an API request asks for result in descending order.
+ * The time sort mapper should be second last mappers to execute on the result set.
```

Comment:
A couple of things here:

1. It's not only activated for descending order, but _any_ order, right?
2. It says that the mapper should be 2nd to last, yet the `add` call in this class seems to be putting it at the end. Something doesn't seem to be correct?